roms_tools/setup/grid.py

```diff
@@ -758,24 +759,24 @@ def from_file(
             "hmin",
         ]:
             if attr in ds.attrs:
-                a = float(ds.attrs[attr])
+                value = float(ds.attrs[attr])
             else:
-                a = None
+                value = None
 
-            object.__setattr__(grid, attr, a)
+            object.__setattr__(grid, attr, value)
 
         if "topography_source_name" in ds.attrs:
             if "topography_source_path" in ds.attrs:
-                a = {
+                topo_source = {
                     "name": ds.attrs["topography_source_name"],
                     "path": ds.attrs["topography_source_path"],
                 }
             else:
-                a = {"name": ds.attrs["topography_source_name"]}
+                topo_source = {"name": ds.attrs["topography_source_name"]}
         else:
-            a = None
+            topo_source = None
 
-        object.__setattr__(grid, "topography_source", a)
+        object.__setattr__(grid, "topography_source", topo_source)
```

Comment:
`topography_source` appears to be a pre-defined attribute of the `Grid` base class. It should be safer if the typechecker can validate the assignment using `grid.topography_source = a`. 

Is there a reason to use `__setattr__`?